src/lib/cache/TTLCache.js

```diff
@@ -0,0 +1,90 @@
+// @flow
+export interface TTLCacheItem<T> {
+  value: T;
+  expire: number;
+}
+
+export type TTLCacheOptions = {
+  ttl: number,
+};
+
+class TTLCache<K, V> {
+  cache: Map<K, TTLCacheItem<V>> = new Map<K, TTLCacheItem<V>>();
+  options: TTLCacheOptions;
+
+  constructor(options?: TTLCacheOptions) {
+    this.options = {
+      ttl: Infinity,
+      ...options,
+    };
+  }
+
+  set(key: K, value: V, ttl: number = this.options.ttl, now: number = Date.now()): boolean {
+    // Adding the value to the cache is not possible if ttl is zero.
+    if (ttl === 0) {
+      return false;
+    }
+
+    // Check for infinity in which case the item persists forever.
+    const expire = ttl < Infinity ? now + ttl : Infinity;
+
+    const item: TTLCacheItem<V> = {
+      value,
+      expire,
+    };
+
+    this.cache.set(key, item);
+
+    return true;
+  }
+
+  get(key: K, remove: boolean = false, now?: number): ?V {
+    const value = this.touch(key, now);
+
+    if (value == null) {
```

Comment:
How about `===`?

Reply:
As explained in slack, this also tests against both `null` and `undefined`.

Reply:
Ah – if this was the intention, a small comment would be cool :)